src/libraries/ERC7739Utils.sol

```diff
@@ -27,30 +25,30 @@ library ERC7739Utils {
     /// @param contentsHash The hash of the contents, per EIP-712
     /// @param domainBytes The encoded domain bytes from EIP-5267
     /// @param appSeparator The app's domain separator
-    /// @param contentsName The top level type, per EIP-712
-    /// @param contentsType The full type string of the contents, per EIP-712
+    /// @param contentsDescr The full type string of the contents, per EIP-712
     function toNestedTypedDataSignHash(
         bytes32 contentsHash,
         bytes memory domainBytes,
         bytes32 appSeparator,
-        string memory contentsName,
-        string memory contentsType
+        string calldata contentsDescr
     ) internal pure returns (bytes32) {
-        bytes32 typedDataSignHash = TypedDataSignLib.hash(contentsName, contentsType, contentsHash, domainBytes);
-        return MessageHashUtils.toTypedDataHash(appSeparator, typedDataSignHash);
+        (string calldata contentsName, string calldata contentsType) = decodeContentsDescr(contentsDescr);
+        if (bytes(contentsName).length == 0 || bytes(contentsType).length == 0) return bytes32(0);
```

Comment:
This prevents reverts if the signature is not encoded correctly for typedDataSign

Reply:
And why do we need to prevent reverts here?

Reply:
Oh because we need it to fallback to other signature verification correct? Can we document that whenever we're returning something empty instead of reverting?

Reply:
Yep, and can do re. documentation